Data/Versions.hs

```diff
@@ -530,53 +524,59 @@ instance Ord Version where
   compare v0@(Version (Just n) _ _) v1@(Version (Just m) _ _) | n == m = compare (wipe v0) (wipe v1)
                                                               | otherwise = compare n m
 
-  -- | If the two Versions were otherwise equal and recursed down this far,
-  -- we need to compare them by their "release" values.
-  compare (Version _ [] rs) (Version _ [] rs') = compare (Version Nothing rs []) (Version Nothing rs' [])
-
-  -- | If one side has run out of chunks to compare but the other hasn't,
-  -- the other must be newer.
-  compare Version{}  (Version _ [] _) = GT
-  compare (Version _ [] _) Version{}  = LT
-
   -- | The usual case. If first VChunks of each Version is equal, then we keep
   -- recursing. Otherwise, we don't need to check further. Consider @1.2@
   -- compared to @1.1.3.4.5.6@.
-  compare (Version _ (a:as) rs) (Version _ (b:bs) rs') = case f a b of
-    EQ  -> compare (Version Nothing as rs) (Version Nothing bs rs')
-    res -> res
-    where f [] [] = EQ
-
-          -- | Opposite of the above. If we've recursed this far and one side
-          -- has fewer chunks, it must be the "greater" version. A Chunk break
-          -- only occurs in a switch from digits to letters and vice versa, so
-          -- anything "extra" must be an @rc@ marking or similar. Consider @1.1@
-          -- compared to @1.1rc1@.
-          f [] _  = GT
-          f _ []  = LT
-
-          -- | The usual case.
-          f (Digits n:ns) (Digits m:ms) | n > m = GT
-                                        | n < m = LT
-                                        | otherwise = f ns ms
-          f (Str n:ns) (Str m:ms) | n > m = GT
-                                  | n < m = LT
-                                  | otherwise = f ns ms
-
-          -- | An arbitrary decision to prioritize digits over letters.
-          f (Digits _ :_) (Str _ :_) = GT
-          f (Str _ :_ ) (Digits _ :_) = LT
+  compare (Version _ as rs) (Version _ bs rs') = g (NEL.toList as) (NEL.toList bs)
```

Comment:
More efficient comparison logic for `Version`. We no longer bother reconstructing a value as we recurse.